.github/workflows/stackhpc-container-image-build.yml

```diff
@@ -207,6 +215,109 @@ jobs:
           path: ${{ matrix.distro }}-container-images
           retention-days: 7
 
+      - name: Generate list of images to scan/push
+        run: |
+          # Clean up any stale data
+          rm -rf ${{ matrix.distro }}-docker-images.txt
+
+          # Make a file of imagename:tag
+          grep --invert-match --no-filename ^REPOSITORY ${{ matrix.distro }}-container-images |\
+          sed 's/ \+/:/g' |\
+          cut -f 1,2 -d: > ${{ matrix.distro }}-docker-images.txt
+        
+      - name: Scan built container images
+        run: |
+          set -euo pipefail
+
+          # Clean any stale data
+          rm -rf ${{ matrix.distro }}-image-scan-output
+          rm -f images-to-push.txt
+
+          # Make a fresh output directory
+          mkdir -p ${{ matrix.distro }}-image-scan-output
+
+          # If Trivy detects no vulnerabilities, add the image name to images-to-push.txt.
+          # If there are vulnerabilities detected, generate a CSV summary and do not add to
+          # images-to-push.txt.
+          while read -r image; do
+            filename=$(basename $image | sed 's/:/\./g')
+            if $(trivy image \
+                    --quiet \
+                    --exit-code 1 \
+                    --scanners vuln \
+                    --format json \
+                    --severity HIGH,CRITICAL \
+                    --output ${{ matrix.distro }}-image-scan-output/${filename}.json \
+                    --ignore-unfixed \
+                    $image); then
+              # Clean up the output file for any images with no vulnerabilities
+              rm -f ${{ matrix.distro }}-image-scan-output/${filename}.json
+
+              # Add the image to the list to push
+              echo "${image}" >> ${{ matrix.distro }}-images-to-push.txt
+            else
+              # Still add the image to the list to push if we're ignoring fails
+              if [ "${{github.event.inputs.scan-push}}" == "true" ]; then
+                echo "${image}" >> ${{ matrix.distro }}-images-to-push.txt
+              fi
+              
+              # Write a header for the summary CSV
+              echo '"PkgName","PkgPath","PkgID","VulnerabilityID","FixedVersion","PrimaryURL","Severity"' > ${{ matrix.distro }}-image-scan-output/${filename}.summary.csv
+
+              # Write the summary CSV data
+              jq -r '.Results[] 
+                     | select(.Vulnerabilities) 
+                     | .Vulnerabilities
+                     # Ignore packages with "kernel" in the PkgName
```

Comment:
Why? kernel seems important...

Reply:
Not sure on this one. I've removed it for now, we can see if it causes problems later down the line